localstack/services/iam/resource_providers/aws_iam_servercertificate.py

```diff
@@ -0,0 +1,129 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class IAMServerCertificateProperties(TypedDict):
+    Arn: Optional[str]
+    CertificateBody: Optional[str]
+    CertificateChain: Optional[str]
+    Path: Optional[str]
+    PrivateKey: Optional[str]
+    ServerCertificateName: Optional[str]
+    Tags: Optional[list[Tag]]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class IAMServerCertificateProvider(ResourceProvider[IAMServerCertificateProperties]):
+    TYPE = "AWS::IAM::ServerCertificate"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[IAMServerCertificateProperties],
+    ) -> ProgressEvent[IAMServerCertificateProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/ServerCertificateName
+
+
+
+        Create-only properties:
+          - /properties/ServerCertificateName
+          - /properties/PrivateKey
+          - /properties/CertificateBody
+          - /properties/CertificateChain
+
+        Read-only properties:
+          - /properties/Arn
+
+        IAM permissions required:
+          - iam:UploadServerCertificate
+          - iam:GetServerCertificate
+
+        """
+        model = request.desired_state
+
```

Comment:
nit: we should add a default for the `ServerCertificateName` since it's not required.